### PR TITLE
Change SMA configuration.toml file to have actual path for Executor.

### LINKED ISSUE #1742: SMA Configuration - Put the Actual Paths in the Config File
Currently there are two fields in the SMA configuration.toml that have a default value which isn't usable. These are as follows:

```
ExecutorPath = '/path/to/the/file'
MetricsMechanism = 'custom/executor/direct-service'
```

Attempting to run the SMA without modifying the configuration to point to the actual executor or metrics mechanism results in an error. These should just point to the out-of-the-box locations so the user doesn't have to waste time troubleshooting something that isn't actually a problem.

Changes need to be made in both the `./res/configuration.toml` and `./res/docker/configuration.toml`

DEFAULT
```
ExecutorPath = '../sys-mgmt-executor/sys-mgmt-executor'
MetricsMechanism = 'direct-service'
```

DOCKER
```
ExecutorPath = '/sys-mgmt-executor'
MetricsMechanism = 'executor'
```

It may be necessary to utilize capabilities from Go's `path/filepath` package in order to make these paths interpretable by Windows.

### PR DESCRIPTION
**Pull request by akramtexas**
Fix #1742 
- Update the SMA's `./res/docker/configuration.toml` file to have the actual path where the Executor is deployed in the image so that it's ready to be used, out-of-the-box, instead of leaving it to the user to populate that: `ExecutorPath = /path/to/the/file'.
- As for DEV_ONLY, the user themself will need to make a corresponding change to the configuration file (i.e. the SMA's `./res/configuration.toml` file) with the appropriate path to match their (local) filesystem. An example provided is called out for ease of setup.
- I did DEV testing by starting out with containerizing an SMA image with Docker. I then deployed the SMA container locally, with all the other Core services (also containerized) running as well. I invoked the functionality of the SMA operations via a REST client (Postman)  to validate that the Executor is deployed in the image so that it's ready to be used, out-of-the-box, instead of leaving it to the user to populate that path via the setting in the SMA's configuration.toml for `ExecutorPath = /path/to/the/file'.
- I exec'd in to the Docker container (for the SMA) and confirmed the filesystem location of the Executor binary (`sys-mgmt-executor`) as in snippet below:
> bash-5.0#
> bash-5.0# pwd
> /
> bash-5.0# ls -al
> ...
> ...
> -rwxr-xr-x    1 root     root      16883529 Sep 18 18:39 sys-mgmt-agent
> -rwxr-xr-x    1 root     root       2901444 Sep 18 18:39 sys-mgmt-executor
> ...
> ...
> 

- Here are some sample results from my DEV testing, starting with a POST operation, using the following (body) payload:

> {
>    "action":"stop",
>    "services":[
>       "edgex-core-data",
>       "edgex-export-distro"
>    ],
>    "params":[
>     "graceful"
>     ]
> }
> 

...and invoking `http://localhost:48090/api/v1/operation` via REST, getting this result:

> [
>     {
>         "Success": true,
>         "executor": "docker",
>         "operation": "stop",
>         "service": "edgex-core-data"
>     },
>     {
>         "Success": true,
>         "executor": "docker",
>         "operation": "stop",
>         "service": "edgex-export-distro"
>     }
> ]

Here's a GET operation, invoking `http://localhost:48090/api/v1/metrics/edgex-core-command` via REST, getting this result:

> [
>     {
>         "Success": true,
>         "executor": "docker",
>         "operation": "metrics",
>         "result": {
>             "cpuUsedPercent": 2.34,
>             "memoryUsed": 5004853,
>             "raw": {
>                 "block_io": "12.7MB / 0B",
>                 "cpu_perc": "2.34%",
>                 "mem_perc": "0.24%",
>                 "mem_usage": "4.773MiB / 1.952GiB",
>                 "net_io": "62.1kB / 43.7kB",
>                 "pids": "14"
>             }
>         },
>         "service": "edgex-core-command"
>     }
> ]

### CODE REVIEW

File: cmd/sys-mgmt-agent/res/configuration.toml [hunk 1]
```diff
@@ -6,15 +6,15 @@
 # In the future, the manifest may be more dynamic or even provided by some 3rd party orchestrator.
 
 ExecutorPath = '/path/to/the/file'
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/configuration.toml]: This line needs to be changed as well to point to the locally built Executor. See proposed example provided in the issue. Needs to be tested and verified.

File: cmd/sys-mgmt-agent/res/configuration.toml [hunk 2]
```diff
@@ -6,15 +6,15 @@
...
-# The preceding line (i.e. '/path/to/the/file') is a placeholder that needs to be populated.
-# The following line (ExecutorPath) is DEV_ONLY, useful-enough to be retained as a comment:
+# The preceding line (i.e. '/path/to/the/file') is a placeholder that needs to be populated for DEV_ONLY.
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/configuration.toml]: This line isn't needed once the ExecutorPath is repointed

File: cmd/sys-mgmt-agent/res/configuration.toml [hunk 3]
```diff
@@ -6,15 +6,15 @@
...
+# Example: The following line (ExecutorPath) is DEV_ONLY. Please modify to match your (local) filesystem:
 # ExecutorPath = '/Users/akramahmad/go/src/github.com/edgexfoundry/edgex-go/cmd/sys-mgmt-executor/sys-mgmt-executor'
 
-MetricsMechanism = 'custom/executor/direct-service'
+# MetricsMechanism = 'custom/executor/direct-service'
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/configuration.toml]: Can be removed

File: cmd/sys-mgmt-agent/res/configuration.toml [hunk 4]
```diff
@@ -6,15 +6,15 @@
...
 # The preceding line (i.e. 'custom/executor/direct-service') is a placeholder that needs to be populated.
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/configuration.toml]: Don't need this line

File: cmd/sys-mgmt-agent/res/docker/configuration.toml [hunk 5]
```diff
@@ -5,15 +5,12 @@
 # In the first release of the SMA, the manifest will be static.
 # In the future, the manifest may be more dynamic or even provided by some 3rd party orchestrator.
 
-ExecutorPath = '/path/to/the/file'
-# The preceding line (i.e. '/path/to/the/file') is a placeholder that needs to be populated.
-# ExecutorPath = '/sys-mgmt-executor'
-
-MetricsMechanism = 'custom/executor/direct-service'
+ExecutorPath = '/sys-mgmt-executor'
+# MetricsMechanism = 'custom/executor/direct-service'
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/docker/configuration.toml]: Don't need this line

File: cmd/sys-mgmt-agent/res/docker/configuration.toml [hunk 6]
```diff
@@ -5,15 +5,12 @@
...
 # The preceding line (i.e. 'custom/executor/direct-service') is a placeholder that needs to be populated.
```
Comment (tsconn23) [inline cmd/sys-mgmt-agent/res/docker/configuration.toml]: Can be removed

**Comment by akramtexas** [comment]
Continuing to troubleshoot why we get this response back, despite having launched the SMA with the following CLI command which should pick up the config setting in `cmd/sys-mgmt-agent/res/configuration.toml`: `./sys-mgmt-agent --confdir=/Users/akramahmad/go/src/github.com/edgexfoundry/edgex-go/cmd/sys-mgmt-agent/res -r`.
```
[
    {
        "operation": "metrics",
        "service": "edgex-core-command",
        "executor": "unknown",
        "Success": false,
        "errorMessage": "fork/exec /path/to/the/file: no such file or directory"
    }
]
```

**Comment by akramtexas** [comment]
- Made the requested updates and retested, both locally and also with Docker. The updated settings in SMA configuration.toml files are picked up correctly by the SMA.

**Pull request merged**